mtags/src/main/scala-2/scala/meta/internal/pc/PcInlayHintsProvider.scala

```diff
@@ -160,23 +167,33 @@ final class PcInlayHintsProvider(
       fun.pos.isOffset && fun.symbol != null && fun.symbol.isImplicit
   }
   object ImplicitParameters {
-    def unapply(tree: Tree): Option[(List[Symbol], Position, Boolean)] =
+    def unapply(
+        tree: Tree
+    ): Option[(List[(Symbol, Boolean)], Position, Boolean)] =
```

Comment:
Maybe it'd be nice to have a dedicated class for `(Symbol, Boolean)`, e.g.
```
sealed trait ImplicitParamSym
case class Param(sym: Symbol) extends ImplicitParamSym
case class ContextBound(sym: Symbol) extends ImplicitParamSym
```
just because Boolean in itself is quite meaningless. But it's up to you.

Reply:
> Maybe:
> 
> ```
> sealed trait ImplicitParamSym
> case class Param(sym: Symbol) extends AnyVal with ImplicitParamSym
> case class ContextBound(sym: Symbol) extends AnyVal with ImplicitParamSym
> ```

This doesn't compile, I used @kasiaMarek solution

Reply:
It does compile for me, what is an error you are getting. I would really use AnyVal in such cases

Reply:
Wait, no:
```

sealed trait ImplicitParamSym extends Any
case class Paramss(sym: Symbol) extends AnyVal with ImplicitParamSym
case class ContextBound(sym: Symbol) extends AnyVal with ImplicitParamSym
```

should work